### PR TITLE
Passenger screen

### LINKED ISSUE #68: Requests to Stop
As a passenger of Pathfinder, I would like to use the stop button so that I may get off the bus.

Acceptance criteria:
* Stop button that toggles on and off upon the click of a button.
* Stop requests will be successfully shown to the driver.

### LINKED ISSUE #86: Accessibility Request
As a passenger of Pathfinder, I would like to have an accessibility request button so that I may indicate that I need more amenities available to me.

Acceptance criteria:
- Accessibility Request button that toggles on and off upon the click of a button.
- Accessibility requests will be successfully shown to the driver.

### PR DESCRIPTION
**Pull request by juliavankirk**
This resolves and closes:
#68 (Requests to Stop)
#86 (Accessibility Requests)

In the passenger screen, there are now Stop and Accessibility request buttons.  They toggle on and off, which displays the relevant request statuses as necessary. 

The boolean data from the toggle buttons get saved as a Shared Preference and is then passed onto the Driver screen, so that drivers are alerted when passengers need to stop, or if they need more  amenities for accessibility if they might be handicapped.

### CODE REVIEW

File: UI/app/src/main/java/com/example/pathfinder/Activities/DriverDashboard.java
Comment (alrapal) [inline UI/app/src/main/java/com/example/pathfinder/Activities/DriverDashboard.java]: Is this a way to declare 2 different variables in one line ? If so, that is pretty cool :p

File: UI/app/src/main/java/com/example/pathfinder/Activities/DriverDashboard.java
Comment (alrapal) [inline UI/app/src/main/java/com/example/pathfinder/Activities/DriverDashboard.java:265]: Why did you deleted this part from the code ? (Just wondering)

File: UI/app/src/main/java/com/example/pathfinder/Activities/DriverDashboard.java
Comment (juliavankirk) [reply UI/app/src/main/java/com/example/pathfinder/Activities/DriverDashboard.java]: Yep! Now that you mention it, I could probably combine some others for better consistency :)

File: UI/app/src/main/java/com/example/pathfinder/Activities/DriverDashboard.java
Comment (juliavankirk) [reply UI/app/src/main/java/com/example/pathfinder/Activities/DriverDashboard.java:265]: I removed it upon request. @gusaxtcha and @iachella wanted to make their own PR to solve the latency issue, which is dependent on this code. 

Please see the conversation below :)
C96CF98E-9CC7-4551-8B8D-E402A50B10D4

Comment (alrapal) [reply UI/app/src/main/java/com/example/pathfinder/Activities/DriverDashboard.java:265]: Ok good! Thanks for the clarification =)

File: UI/app/src/main/java/com/example/pathfinder/Activities/DriverDashboard.java
Comment (alrapal) [reply UI/app/src/main/java/com/example/pathfinder/Activities/DriverDashboard.java]: Cool, I did not know we could do that. Good to know 😄

**Review by alrapal — COMMENTED** [review]
Just one last question, the shared preferences you pass to the driver side, does that work if the app is launched on 2 different devices?

**Comment by juliavankirk** [comment]
@alrapal no, Shared Preference is a primitive way of storing and passing data between two different activities.  The group decided against implementing a database, which limits any sort of way to communicate between multiple processes.

**Comment by alrapal** [comment]
> @alrapal no, Shared Preference is a primitive way of storing and passing data between two different activities. The group decided against implementing a database, which limits any sort of way to communicate between multiple processes.

Ok, I see.
Yes, I know about the DB, but maybe there is a way to use MQTT to pass it ? (Just throwing ideas)

**Comment by juliavankirk** [comment]
@alrapal It wouldn't be optimal to pass it via MQTT since we are already dealing with a large amount of backlog due to latency issues.

**Comment by alrapal** [comment]
> @alrapal It wouldn't be optimal to pass it via MQTT since we are already dealing with a large amount of backlog due to latency issues.

I am not sure it would impact MQTT that much. Especially since the optimisation #58 is on its way [EDIT: precisions regarding optimisation]. Although it might. 

My only concern with this solution is that passengers and drivers are not communicating information "for real". Which is fine if we say we assume they are. I guess we could ask the rest of the group to have more insights about this. 
In reality we would not use MQTT for these kind of things anyway, especially not for the video. 

I have no strong feelings for neither but I think I would need the opinion of others before approving if that's alright =)

**Review by iachella — CHANGES_REQUESTED** [review]
The code on the xml side looks good.
Regarding the conversation about stop requested and accessibility using shared preferences, I understand that it might affect the latency.
However, since these features imply a small amount of messages being sent, and we don't have any latency issues with the code that will be implemented with #58 , we could find out that using mqtt instead would not bring any noticeable lagging, while bringing value to the implemented issues since we won't need to have assumptions about using different devices.

We can use tomorrow's meeting to clarify this with the others before merging :)

**Comment by bassamEldesouki** [comment]
We also need to consider that in real life it won't be just one passenger clicking this stop button so for sure the Latency will be effected, so I understand Julia's point. However in our case right now, I would consider this is as first version of the app and there are for sure many things to improve later on, between them the latency! Therefor I would think of passing the info through MQTT as a temporary solution for this first version. It won't be effected because not many messages will be sent. But then we will need to modify our requirements list and inform the customer that this will need to be improved.

**Review by iachella — APPROVED** [review]
After the discussion we had in this afternoon's meeting, we agreed that the way this code is implemented is okay to go for master. Further improvements might follow.

**Pull request merged**
